staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -3682,7 +3683,7 @@ func (c *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, apiS
 			}
```

Comment:
If `port.Protocol` is UDP, it doesn't make sense to register that port for TCP health-checks. And AWS NLB only offers the options of [HTTP, HTTPS, and TCP](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/target-group-health-checks.html#health-check-settings).

Reply:
Perhaps we can leave these unset in the UDP case, and catch that later and fail the NLB creation if `externalTrafficPolicy` is not set to `Local`. At least for a first iteration, it'd be an improvement on where we are now.

Reply:
Currently, AWS NLB doesn't support disabling the health check completely. Since the ( protocol, port) values are required, default (TCP, traffic-port) will get used if `externalTrafficPolicy` is set to `Cluster`. This will at least provide an option to run an auxiliary service that responds to the LB health queries.

Reply:
Is that worth the trade-off for surprising "LB refuses to send you traffic" results when there _isn't_ such an auxiliary service?

Reply:
If all registered targets are unhealthy, NLB routes the requests across the unhealthy targets. Since a functional LB is still possible, IMO it doesn't add much value to fail the lb creation.

This does provide us an opportunity to have additional parameters for the health check configuration in future.

Reply:
https://github.com/kubernetes/kubernetes/issues/92321